src/main/java/org/broadinstitute/hellbender/tools/walkers/annotator/VariantAnnotatorEngine.java

```diff
@@ -0,0 +1,263 @@
+package org.broadinstitute.hellbender.tools.walkers.annotator;
+
+import com.google.common.collect.Sets;
+import htsjdk.variant.variantcontext.*;
+import htsjdk.variant.vcf.*;
+import org.broadinstitute.hellbender.engine.FeatureContext;
+import org.broadinstitute.hellbender.engine.FeatureInput;
+import org.broadinstitute.hellbender.engine.ReferenceContext;
+import org.broadinstitute.hellbender.exceptions.UserException.BadArgumentValue;
+import org.broadinstitute.hellbender.tools.walkers.annotator.interfaces.AnnotationGroup;
+import org.broadinstitute.hellbender.tools.walkers.annotator.interfaces.GenotypeAnnotation;
+import org.broadinstitute.hellbender.tools.walkers.annotator.interfaces.InfoFieldAnnotation;
+import org.broadinstitute.hellbender.tools.walkers.annotator.interfaces.VariantAnnotation;
+import org.broadinstitute.hellbender.utils.Utils;
+import org.broadinstitute.hellbender.utils.genotyper.PerReadAlleleLikelihoodMap;
+
+import java.util.*;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static org.broadinstitute.hellbender.tools.walkers.annotator.interfaces.AnnotationGroup.ActiveRegionBased;
+import static org.broadinstitute.hellbender.tools.walkers.annotator.interfaces.AnnotationGroup.Standard;
+
+/**
+ * The class responsible for computing annotations for variants.
+ */
+public final class VariantAnnotatorEngine {
+    private final List<InfoFieldAnnotation> infoAnnotations;
+    private final List<GenotypeAnnotation> genotypeAnnotations;
+
+    private final VariantOverlapAnnotator variantOverlapAnnotator;
+
+    private VariantAnnotatorEngine(final AnnotationManager annots, final FeatureInput<VariantContext> dbSNPBinding){
+        infoAnnotations = annots.createInfoFieldAnnotations();
+        genotypeAnnotations = annots.createGenotypeAnnotations();
+        variantOverlapAnnotator = initializeDBs(dbSNPBinding);
+    }
+
+    /**
+     * Makes the engine for all known annotation types (minus the excluded ones).
+     */
+    public static VariantAnnotatorEngine ofAll(final List<String> annotationsToExclude, final FeatureInput<VariantContext> dbSNPBinding) {
+        Utils.nonNull(annotationsToExclude, "annotationsToExclude is null");
+        return new VariantAnnotatorEngine(AnnotationManager.ofAll(annotationsToExclude), dbSNPBinding);
+    }
+
+    /**
+     * Makes the engine for given annotation types and groups (minus the excluded ones).
+     */
+    public static VariantAnnotatorEngine ofSelected(final List<String> annotationGroupsToUse, final List<String> annotationsToUse, final List<String> annotationsToExclude, final FeatureInput<VariantContext> dbSNPBinding) {
```

Comment:
I don't understand why there is an ofAll _and_ an ofSelected.  Seems like some code duplication could be avoided here.